lib/sinon/stub-descriptor.js

```diff
@@ -0,0 +1,43 @@
+"use strict";
+
+var deprecated = require("./util/core/deprecated");
+var objectKeys = require("./util/core/object-keys");
+var spy = require("./spy");
+var wrapMethod = require("./util/core/wrap-method");
+
+// This is deprecated and will be removed in a future version of sinon.
+// We will only consider pull requests that fix serious bugs in the implementation
+function stubDescriptor(object, property, descriptor) {
+    var wrapper;
+
+    deprecated.printWarning(
+      "sinon.stub(obj, 'meth', fn) is deprecated and will be removed from" +
```

Comment:
@lucasfcosta that's an interesting discussion. I am not sure it should block merging of this PR. If you agree, would you mind opening a separate issue for the discussion, so we can merge this?